test/e2e/tests/metrics/delete-metametrics-data.spec.ts

```diff
@@ -0,0 +1,261 @@
+import { Suite } from 'mocha';
+import {
+  defaultGanacheOptions,
+  withFixtures,
+  getEventPayloads,
+  unlockWallet,
+} from '../../helpers';
+import FixtureBuilder from '../../fixture-builder';
+import { Driver } from '../../webdriver/driver';
+import { TestSuiteArguments } from '../confirmations/transactions/shared';
+import { WebElementWithWaitForElementState } from '../../webdriver/types';
+
+const selectors = {
+  accountOptionsMenuButton: '[data-testid="account-options-menu-button"]',
+  globalMenuSettingsButton: '[data-testid="global-menu-settings"]',
+  securityAndPrivacySettings: { text: 'Security & privacy', tag: 'div' },
+  experimentalSettings: { text: 'Experimental', tag: 'div' },
+  deletMetaMetricsSettings: '[data-testid="delete-metametrics-data-button"]',
+  deleteMetaMetricsDataButton: {
+    text: 'Delete MetaMetrics data',
+    tag: 'button',
+  },
+  clearButton: { text: 'Clear', tag: 'button' },
+  backButton: '[data-testid="settings-back-button"]',
+};
+
+/**
+ * mocks the segment api multiple times for specific payloads that we expect to
+ * see when these tests are run. In this case we are looking for
+ * 'Permissions Requested' and 'Permissions Received'. Do not use the constants
+ * from the metrics constants files, because if these change we want a strong
+ * indicator to our data team that the shape of data will change.
+ *
+ * @param mockServer
+ * @returns
+ */
+const mockSegment = async (mockServer: Mockttp) => {
+  return [
+    await mockServer
+      .forPost('https://api.segment.io/v1/batch')
+      .withJsonBodyIncluding({
+        batch: [
+          { type: 'track', event: 'Delete MetaMetrics Data Request Submitted' },
+        ],
+      })
+      .thenCallback(() => {
+        return {
+          statusCode: 200,
+        };
+      }),
+    await mockServer
+      .forPost('https://metametrics.metamask.test/regulations/sources/test')
+      .withHeaders({ 'Content-Type': 'application/vnd.segment.v1+json' })
+      .withBodyIncluding(
+        JSON.stringify({
+          regulationType: 'DELETE_ONLY',
+          subjectType: 'USER_ID',
+          subjectIds: ['fake-metrics-id'],
+        }),
+      )
+      .thenCallback(() => ({
+        statusCode: 200,
+        json: { data: { regulateId: 'fake-delete-regulation-id' } },
+      })),
+    await mockServer
+      .forGet(
+        'https://metametrics.metamask.test/regulations/fake-delete-regulation-id',
+      )
+      .withHeaders({ 'Content-Type': 'application/vnd.segment.v1+json' })
+      .thenCallback(() => ({
+        statusCode: 200,
+        json: {
+          data: {
+            regulation: {
+              overallStatus: 'FINISHED',
+            },
+          },
+        },
+      })),
+  ];
+};
+/**
+ * Scenarios:
+ * 1. Deletion while Metrics is Opted in.
+ * 2. Deletion while Metrics is Opted out.
+ * 3. Deletion when user never opted for metrics.
+ */
+describe('Delete MetaMetrics Data @no-mmi', function (this: Suite) {
```

Comment:
Nit: Is it a lot of effort to migrate this to POM e2e tests pattern? We can also leave it a TODO to be targeted in another PR

Reply:
Since we aim to get it merged as soon as possible, I would prefer to add a TODO and create a ticket for the same.